source/dub/packagesupplier.d

```diff
@@ -254,4 +267,41 @@ class RegistryPackageSupplier : PackageSupplier {
 	}
 }
 
+abstract class AbstractFallbackPackageSupplier : PackageSupplier
+{
+	protected PackageSupplier m_default, m_fallback;
```

Comment:
I'd hope though that a third site wouldn't be necessary. We're also far from needing some round-robing load balancing, and should look at server side reliability/DNS etc. if that ever becomes a topic.